### PR TITLE
Revert argsort

### PR DESCRIPTION
**Pull request by mlondschien**

### CODE REVIEW

**Comment by mlondschien** [comment]
```
 Benchmarking argsort/argsort_continuous/1000
Benchmarking argsort/argsort_continuous/1000: Warming up for 3.0000 s
Benchmarking argsort/argsort_continuous/1000: Collecting 100 samples in estimated 5.1252 s (131k iterations)
Benchmarking argsort/argsort_continuous/1000: Analyzing
argsort/argsort_continuous/1000
                        time:   [38.816 us 38.836 us 38.856 us]
                        change: [+54.083% +54.265% +54.447%] (p = 0.00 < 0.05)
                        Performance has regressed.
Benchmarking argsort/argsort_one_hot/1000
Benchmarking argsort/argsort_one_hot/1000: Warming up for 3.0000 s
Benchmarking argsort/argsort_one_hot/1000: Collecting 100 samples in estimated 5.0180 s (783k iterations)
Benchmarking argsort/argsort_one_hot/1000: Analyzing
argsort/argsort_one_hot/1000
                        time:   [6.4022 us 6.4032 us 6.4044 us]
                        change: [-60.634% -60.614% -60.599%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 12 outliers among 100 measurements (12.00%)
  1 (1.00%) low severe
  1 (1.00%) low mild
  5 (5.00%) high mild
  5 (5.00%) high severe
Benchmarking argsort/argsort_continuous/10000
Benchmarking argsort/argsort_continuous/10000: Warming up for 3.0000 s
Benchmarking argsort/argsort_continuous/10000: Collecting 100 samples in estimated 7.3801 s (10k iterations)
Benchmarking argsort/argsort_continuous/10000: Analyzing
argsort/argsort_continuous/10000
                        time:   [731.06 us 731.11 us 731.16 us]
                        change: [-5.4706% -5.4464% -5.4258%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 8 outliers among 100 measurements (8.00%)
  1 (1.00%) low severe
  6 (6.00%) high mild
  1 (1.00%) high severe
Benchmarking argsort/argsort_one_hot/10000
Benchmarking argsort/argsort_one_hot/10000: Warming up for 3.0000 s
Benchmarking argsort/argsort_one_hot/10000: Collecting 100 samples in estimated 5.1595 s (66k iterations)
Benchmarking argsort/argsort_one_hot/10000: Analyzing
argsort/argsort_one_hot/10000
                        time:   [78.574 us 78.581 us 78.588 us]
                        change: [-70.776% -70.770% -70.765%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 8 outliers among 100 measurements (8.00%)
  1 (1.00%) low severe
  3 (3.00%) low mild
  2 (2.00%) high mild
  2 (2.00%) high severe
Benchmarking argsort/argsort_continuous/100000
Benchmarking argsort/argsort_continuous/100000: Warming up for 3.0000 s
Benchmarking argsort/argsort_continuous/100000: Collecting 100 samples in estimated 5.4686 s (600 iterations)
Benchmarking argsort/argsort_continuous/100000: Analyzing
argsort/argsort_continuous/100000
                        time:   [9.1055 ms 9.1127 ms 9.1250 ms]
                        change: [-11.213% -11.126% -10.991%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  6 (6.00%) high mild
  3 (3.00%) high severe
Benchmarking argsort/argsort_one_hot/100000
Benchmarking argsort/argsort_one_hot/100000: Warming up for 3.0000 s

Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 5.7s, enable flat sampling, or reduce sample count to 60.
Benchmarking argsort/argsort_one_hot/100000: Collecting 100 samples in estimated 5.6598 s (5050 iterations)
Benchmarking argsort/argsort_one_hot/100000: Analyzing
argsort/argsort_one_hot/100000
                        time:   [1.1205 ms 1.1206 ms 1.1208 ms]
                        change: [-73.088% -73.079% -73.069%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 13 outliers among 100 measurements (13.00%)
  3 (3.00%) low mild
  2 (2.00%) high mild
  8 (8.00%) high severe
Benchmarking argsort/argsort_continuous/1000000
Benchmarking argsort/argsort_continuous/1000000: Warming up for 3.0000 s

Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 11.5s, or reduce sample count to 40.
Benchmarking argsort/argsort_continuous/1000000: Collecting 100 samples in estimated 11.457 s (100 iterations)
Benchmarking argsort/argsort_continuous/1000000: Analyzing
argsort/argsort_continuous/1000000
                        time:   [115.52 ms 115.69 ms 115.86 ms]
                        change: [-3.0403% -2.8696% -2.6805%] (p = 0.00 < 0.05)
                        Performance has improved.
Benchmarking argsort/argsort_one_hot/1000000
Benchmarking argsort/argsort_one_hot/1000000: Warming up for 3.0000 s
Benchmarking argsort/argsort_one_hot/1000000: Collecting 100 samples in estimated 5.2662 s (600 iterations)
Benchmarking argsort/argsort_one_hot/1000000: Analyzing
argsort/argsort_one_hot/1000000
                        time:   [8.7645 ms 8.7677 ms 8.7710 ms]
                        change: [-80.898% -80.866% -80.834%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 2 outliers among 100 measurements (2.00%)
  2 (2.00%) high mild
```

TLDR: Worse for small sample (`n = 1000`) continuous. Faster everywhere else.
